### PR TITLE
fixes to cover multistep and tests

### LINKED ISSUE #493: CoverMultistepOptions environment appears to find invalid solutions
I changed the hand regions in the environment, but solutions seem to allow the robot to place outside of the changed hand regions:

https://user-images.githubusercontent.com/8839164/155862505-3d29193f-0ba6-49c3-b161-7fa18212ec18.mp4

### PR DESCRIPTION
**Pull request by tomsilver**
closes #493 and should resolve the unit test failures in #479 

the issue in #493 was an edge case where the hand is placing outside of both the allowed hand region _and_ outside of the target region. before this PR, the code would think that this placement was a "place on table", which is allowed. but that's not always the case, like in this video:

https://user-images.githubusercontent.com/2816502/156375704-ed0135e9-1f9e-467a-9080-e93cc85f74df.mp4

the unit test failures in #479 were related to default hyperparameter settings being challenging for oracle planning. we want to leave the defaults alone because we want the environment to be challenging, but I made them easier in the relevant unit tests.

### CODE REVIEW

**Review by ronuchit — COMMENTED** [review]
lgtm, should wait for @ashayathalye 's approval

File: tests/envs/test_cover.py
Comment (ronuchit) [inline tests/envs/test_cover.py:569]: if keeping this, you can move the two below `pragma: no cover` to just being on this `if` statement, so there's just one

File: tests/envs/test_cover.py
Comment (tomsilver) [reply tests/envs/test_cover.py:565]: they are useful, for example the way that I made the new unit test was by first making a video for one of the existing ones to remember what it does, and the modifying the action sequence and rendering videos along the way

File: tests/envs/test_cover.py
Comment (tomsilver) [reply tests/envs/test_cover.py]: did that

**Review by NishanthJKumar — COMMENTED** [review]
LGTM!

**Review by ronuchit — APPROVED** [review]

**Pull request merged**
